yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -176,3 +176,153 @@ def test_annotations(self, query_tracker):
+
+    @authors("krock21")
+    def test_access_control_simple_case(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_access_control_object(
+            "aco1",
+            "queries",
+            attributes={
+                "acl": [
+                    make_ace("allow", "u2", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco1")
+        q_u2 = start_query("mock", "run_forever", authenticated_user="u2", access_control_object="aco2")
+        q_u1.get(authenticated_user="u2")
+        with raises_yt_error(500):  # Node "aco2" not found.
+            q_u2.get(authenticated_user="u1")
+
+    @authors("krock21")
+    def test_access_control_get(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("u3")
+        create_access_control_object(
+            "aco_get",
+            "queries",
+            attributes={
+                "acl": [
+                    make_ace("allow", "u2", "use"),
+                    make_ace("allow", "u3", "read"),
+                    make_ace("allow", "u3", "write"),
+                    make_ace("allow", "u3", "administer"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_get")
+        q_u1.get(authenticated_user="u2")
+        with raises_yt_error(3903):  # Access Denied.
+            q_u1.get(authenticated_user="u3")
+
+    @authors("krock21")
+    def test_access_control_abort(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("u3")
+        create_access_control_object(
+            "aco_abort",
+            "queries",
+            attributes={
+                "acl": [
+                    make_ace("allow", "u2", "administer"),
+                    make_ace("allow", "u3", "read"),
+                    make_ace("allow", "u3", "write"),
+                    make_ace("allow", "u3", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_abort")
+        q_u1.abort(authenticated_user="u2")
+        with raises_yt_error(3903):  # Access Denied.
+            q_u1.abort(authenticated_user="u3")
+
+    @authors("krock21")
+    def test_access_control_get_query_result(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("u3")
+        create_access_control_object(
+            "aco_get_query_result",
+            "queries",
+            attributes={
+                "acl": [
+                    make_ace("allow", "u2", "read"),
+                    make_ace("allow", "u3", "administer"),
+                    make_ace("allow", "u3", "write"),
+                    make_ace("allow", "u3", "use"),
+                    make_ace("deny", "u3", "read"),  # doesnt work without it
```

Comment:
Since I used `acl` instead of `principal_acl`, default "read" permission was inherited from root for everyone.

Switched to `principal_acl` and removed this line